lib/install/management/store_files.go

```diff
@@ -72,19 +71,44 @@ func (d *Dispatcher) DeleteStores(vchVM *vm.VirtualMachine, conf *config.Virtual
 	return nil
 }
 
-func (d *Dispatcher) deleteParent(ds *object.Datastore, root string) error {
-	defer trace.End(trace.Begin(""))
+func (d *Dispatcher) deleteImages(conf *config.VirtualContainerHostConfigSpec) error {
+	var emptyImages bool
+	var errs []string
+
+	for _, imageDir := range conf.ImageStores {
+		imageDSes, err := d.session.Finder.DatastoreList(d.ctx, imageDir.Host)
+		if err != nil {
+			errs = append(errs, err.Error())
+			continue
+		}
+
+		if len(imageDSes) != 1 {
+			errs = append(errs, fmt.Sprintf("Invalid or ambiguous datastore name %s provided while attempting to remove image stores", imageDir.Host))
```

Comment:
I'm not a UX expert, but these kinds of error messages are infuriating because they tell me little about the actual condition that's throwing the error.  How about we let the user know that the datastore provided maps to >1 datastores found?